test/dynamics/solvers/test_solver_functions.py

```diff
@@ -234,25 +234,71 @@ def func(a):
 class TestRK4(TestSolverMethod):
     """Test class for RK4_solver."""
 
-    def solve(self, rhs, t_span, y0, t_eval=None, **kwargs):
-        return solve_ode(
-            rhs=rhs, t_span=t_span, y0=y0, method="RK4", t_eval=t_eval, max_dt=0.001, **kwargs
-        )
+    def solve(self, rhs, t_span, y0, t_eval=None, solver_func="ode", **kwargs):
```

Comment:
For this `solver_func` should just straight up by the callable `solve_lmde` or `solve_ode`, then for the body of the function you can do:

```
     return solver_func(
         rhs,
         t_span=t_span,
         y0=y0,
         method="RK4",
         t_eval=t_eval,
         max_dt=0.001,
         **kwargs,
     )
```